tests/subworkflows/create_test_yml.py

```diff
@@ -0,0 +1,96 @@
+import os
+from pathlib import Path
+from unittest import mock
+
+import pytest
+
+import nf_core.subworkflows
+
+from ..utils import with_temporary_folder
```

Comment:
We should start using pytest's [`tmp_path`](https://docs.pytest.org/en/stable/how-to/tmp_path.html) fixture instead of our custom cooked solution.

Reply:
I don't think that will work, unless we completely refactor our tests to be fully based on `pytest` instead of `unittest` (https://docs.pytest.org/en/7.1.x/how-to/unittest.html#pytest-features-in-unittest-testcase-subclasses)

Reply:
It was more a general comment, rather than something I expected to see in this PR 🌱